### PR TITLE
feat: [IA-668] Open the email client instead of the web form if the user denies submitting their information

### PR DESCRIPTION
**Pull request by debiff**
## Short description
This PR aims to open the email client instead of the web form if the user denies submitting their information when opening a support ticket.

This PR also changes the screens' order in the Zendesk support workflow as a side effect.
Now, we first ask the user the ticket category then ask for permission to send the information.

## List of changes proposed in this pull request
- Added the `selectedSubcategory` reducer 
- Modified the order of the screens
- Open the predefined email client with the category (and eventually the subcategory) as subject

https://user-images.githubusercontent.com/11773070/156548222-0a20a352-1913-42b0-890f-03f46c98fef6.MP4

## How to test
The test suite was updated.
To test the email client opening you must use the physical device since there isn't an email client in the simulators.

### CODE REVIEW

**Comment by thisisjp** [comment]
@debiff can we add a space between the ":" and the sub-category? 🤓

File: ts/features/zendesk/screens/ZendeskAskPermissions.tsx [hunk 1]
```diff
@@ -236,7 +247,12 @@ const ZendeskAskPermissions = (props: Props) => {
     .filter(it => it.value !== notAvailable);
 
   const handleOnCancel = () => {
-    openWebUrl(assistanceWebFormLink);
+    handleItemOnPress(
```
Comment (Undermaken) [inline ts/features/zendesk/screens/ZendeskAskPermissions.tsx:250]: What about the error scenario? 
If the users have no email client set, should we show a message error?

https://user-images.githubusercontent.com/822471/157081720-e28ef126-5f75-41cb-97a2-8eeef0ffd067.mov

**Review by Undermaken — CHANGES_REQUESTED** [review]

File: ts/features/zendesk/components/ZendeskSupportComponent.tsx [hunk 2]
```diff
@@ -107,13 +108,23 @@ const ZendeskSupportComponent = (props: Props) => {
...
+    }
+
+    if (canSkipCategoryChoice) {
       navigation.navigate(
         navigateToZendeskAskPermissions({ assistanceForPayment })
       );
+    } else {
+      navigation.navigate(
+        navigateToZendeskChooseCategory({ assistanceForPayment })
+      );
```
[the remark this answers is not in the document]
Comment (debiff) [reply ts/features/zendesk/components/ZendeskSupportComponent.tsx]: Thank you, refactored in [dbb4b77](https://github.com/pagopa/io-app/pull/3801/commits/dbb4b773f2f927360dc9c418b279bc5d6e14fa0f)

File: ts/features/zendesk/screens/ZendeskAskPermissions.tsx [hunk 3]
```diff
@@ -236,7 +247,12 @@ const ZendeskAskPermissions = (props: Props) => {
...
+      anonymousAssistanceAddress(
+        zendeskSelectedCategory.description["it-IT"],
+        zendeskSelectedSubcategory?.description["it-IT"]
+      )
+    )();
```
[the remark this answers is not in the document]
Comment (debiff) [reply ts/features/zendesk/screens/ZendeskAskPermissions.tsx]: Yes, fixed in [181ae78](https://github.com/pagopa/io-app/pull/3801/commits/181ae78d102a180409013165b7a5638363a2ac3b)

**Review by Undermaken — APPROVED** [review]

**Pull request merged**